test/jit/test_schema_check.py

```diff
@@ -96,3 +142,26 @@ def test_mutation_check_fail_multiple_operators(self):
                 x.tanh_()
                 x.relu_()
                 batch(SchemaCheckMode(x))
+
+    # Tests that an exception is raised for a mismatching alias
+    def test_alias_check_fail(self):
+        with self.assertRaises(RuntimeError):
+            x = torch.rand((3, 3), requires_grad=True)
+            y = torch.rand((3, 3), requires_grad=True)
+            with enable_torch_dispatch_mode(SchemaCheckMode()):
+                IncorrectAliasTensor(x).add(IncorrectAliasTensor(y), alpha=2)
+
+    def test_alias_check_fail_multiple_operators(self):
+        with self.assertRaises(RuntimeError):
+            x = torch.rand((3, 3), requires_grad=True)
+            y = torch.rand((3, 3), requires_grad=True)
+            with enable_torch_dispatch_mode(SchemaCheckMode()):
+                IncorrectAliasTensor(x).sin().relu().add(IncorrectAliasTensor(y), alpha=2)
```

Comment:
nit: just for completeness could you add a test where add is one of the ops in the middle?